crates/rpc/rpc-types/src/eth/engine/payload.rs

```diff
@@ -122,78 +122,77 @@ impl From<SealedBlock> for ExecutionPayload {
     }
 }
 
-/// Try to construct a block from given payload. Perform addition validation of `extra_data` and
-/// `base_fee_per_gas` fields.
-///
-/// NOTE: The log bloom is assumed to be validated during serialization.
-/// NOTE: Empty ommers, nonce and difficulty values are validated upon computing block hash and
-/// comparing the value with `payload.block_hash`.
-///
-/// See <https://github.com/ethereum/go-ethereum/blob/79a478bb6176425c2400e949890e668a3d9a3d05/core/beacon/types.go#L145>
-impl TryFrom<ExecutionPayload> for SealedBlock {
-    type Error = PayloadError;
-
-    fn try_from(payload: ExecutionPayload) -> Result<Self, Self::Error> {
-        if payload.extra_data.len() > MAXIMUM_EXTRA_DATA_SIZE {
-            return Err(PayloadError::ExtraData(payload.extra_data))
+impl ExecutionPayload {
+    /// Tries to create a new block from the given payload and optional parent beacon block root.
+    /// Perform additional validation of `extra_data` and `base_fee_per_gas` fields.
+    ///
+    /// NOTE: The log bloom is assumed to be validated during serialization.
+    /// NOTE: Empty ommers, nonce and difficulty values are validated upon computing block hash and
+    /// comparing the value with `payload.block_hash`.
+    ///
+    /// See <https://github.com/ethereum/go-ethereum/blob/79a478bb6176425c2400e949890e668a3d9a3d05/core/beacon/types.go#L145>
+    pub fn try_into_sealed_block(
+        self,
+        parent_beacon_block_root: Option<H256>,
+    ) -> Result<SealedBlock, PayloadError> {
```

Comment:
meh - can we/should we capture this in executionpayload? i g not?

Reply:
The API defines it as a separate argument, so if we were to put it into the execution payload we'd have the field duplicated across the execution payload and RPC args, which isn't ideal:

> * method: `engine_newPayloadV3`
> * params:
>   1. `executionPayload`: [`ExecutionPayloadV3`](https://github.com/ethereum/execution-apis/blob/main/src/engine/cancun.md#executionpayloadv3).
>   2. `expectedBlobVersionedHashes`: `Array of DATA`, 32 Bytes - Array of expected blob versioned hashes to validate.
>   3. `parentBeaconBlockRoot`: `DATA`, 32 Bytes - Root of the parent beacon block.

@mattsse was referencing a wrapper type for this
> I thought that we could introduce a new wrapper type here, but I actually prefer having a Some|None argument in all the on_new_payload functions so it is evident when we're using it

which would wrap the execution payload and be used as the type in the newpayload message, and arguments for `new_payload`, `ensure_well_formed_payload`, etc